### ISSUE TITLE
Events not generating for BaseCmd classes

### ISSUE THREAD

**Issue by vishesh92** [apache/cloudstack#8771]
##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~
all
~~~

##### SUMMARY

Events are not being generated for commands which extend BaseCmd. This results in missing events for some commands.

Ref.: https://github.com/apache/cloudstack/pull/8630#pullrequestreview-1914158484

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by shwstppr** [comment]
In my understanding, action events in ACS are created mainly by the `@ActionEvent` annotation. For `BaseAsyncCmd` we do publish events with `Scheduled` state but that is just it. Trying to publish events for `BaseCmd` may require a lot of changes and refactoring. Would it be better to fix the specific cases where events are not generated highlighted in https://github.com/apache/cloudstack/pull/8630#pullrequestreview-1914158484? 
@vishesh92 @rohityadavcloud @sureshanaparti

**Comment by harikrishna-patnala** [comment]
> In my understanding, action events in ACS are created mainly by the `@ActionEvent` annotation. For `BaseAsyncCmd` we do publish events with `Scheduled` state but that is just it. Trying to publish events for `BaseCmd` may require a lot of changes and refactoring. Would it be better to fix the specific cases where events are not generated highlighted in [#8630 (review)](https://github.com/apache/cloudstack/pull/8630#pullrequestreview-1914158484)? [@vishesh92](https://github.com/vishesh92) [@rohityadavcloud](https://github.com/rohityadavcloud) [@sureshanaparti](https://github.com/sureshanaparti)

Agreed with @shwstppr comment, it is mainly addressed by ActionEvent annotation. I prefer to fix the specific cases. @vishesh92 are you fine with this ?

**Comment by vishesh92** [comment]
> Agreed with [@shwstppr](https://github.com/shwstppr) comment, it is mainly addressed by ActionEvent annotation. I prefer to fix the specific cases. [@vishesh92](https://github.com/vishesh92) are you fine with this ?

I was thinking of a standard way in order to ensure we don't miss it for new commands added in the future. If the effort seems a lot, I am okay with specific cases as well.

**Comment by DaanHoogland** [comment]
I'm not sure how complicated this will be but it seems mostly search work per event, so @vishesh92 let's make seperate child issues for
- hosts
- projects
- internal lb
- ssh key pairs
- affinity groups
- vm snapshots
- firewall
- static nat

as per @kiranchavala 's report in https://github.com/apache/cloudstack/pull/8630#pullrequestreview-1914158484

makes sense?

**Comment by harikrishna-patnala** [comment]
fixed few cases mentioned in the comments by the PR https://github.com/apache/cloudstack/pull/11230
